### ISSUE TITLE
Feature request: Redis cluster mode enabled => ensure AutomaticFailover is enabled.

### ISSUE THREAD

**Issue by sbol-coolblue** [aws-cloudformation/cfn-python-lint#447]
Found an issue which can probably easily be picked up by the linter.

When creating a ElastiCacheReplicationGroup, if the replication group has a ElastiCacheParameterGroup with parameter 'cluster-enabled' set to 'yes', then the ElastiCacheReplicationGroup must have its `AutomaticFailoverEnabled` property set to true. Failing to do so yields `Redis with cluster mode enabled cannot be created with auto failover turned off. (Service: AmazonElastiCache; Status Code: 400; Error Code: InvalidParameterValue)`.

**Comment by cmmeyer** [comment]
Thanks for suggestion!  Do you know if this  Is this Redis specific or does it apply to Memcached as well?

**Comment by sbol-coolblue** [comment]
This one is Redis-specific, because the ElasticacheReplicationGroup is used by Redis only. 

Related, but perhaps already covered, is the following: "If you specify true, you must specify a value greater than 1 for the NumCacheClusters property. "

So in pseudo-code:
Within a Cache Cluster, if the ElastiCacheParameterGroup::parameters contains "cluster-enabled", and the value of that param is true, then ElasticacheReplicationGroup::AutomaticFalloverEnabled must be "true-or-missing".

if property "AutomaticFailoverEnabled" is "true-or-missing"
Then Check NumCacheClusters > 1

True-or-missing is due to it defaulting to true.

**Comment by kddejong** [comment]
This is waiting on some logic around how to compare two resources and the conditions that may affect their properties.

**Comment by kddejong** [comment]
I started working on this one but there a few other things to note.

```
You cannot enable automatic failover for Redis versions earlier than 2.8.6 or for T1 cache node types. Automatic failover is supported on T2 node types only if you are running Redis version 3.2.4 or later with cluster mode enabled.
```
